src/components/DateSelector/DateSelector.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from 'react'
+import styles from './DateSelector.styles'
+import Select from './Select'
+// import '@resmio/rollico'
+
+import {
+  isValidDate,
+  getDayFromDate,
+  getMonthFromDate,
+  getPastYears,
+  getYearFromDate,
+  MONTH_NAMES
+} from '../../utils/dates'
+
+// Date Generation
+const days = [...Array(31 + 1).keys()].slice(1)
+const actualYear = getYearFromDate(new Date())
+const years = getPastYears(120, actualYear)
+
+// Validation
+const validate = date => {
+  const errors = {}
+  if (date.day === 0) { errors.day = true }
+  if (date.month === 0) { errors.month = true }
+  if (date.year === 0) { errors.year = true }
+  if (!isValidDate(date)) { errors.invalid = true }
+  return errors
+}
+
+// Component
+//  - Props:
+//      - date :Date
+//      - monthBeforeDay :Boolean
+//      - onChange :Func
+
+class DateSelector extends Component {
+  state = {
+    date: {
+      month: -1,
+      day: 0,
+      year: 0
+    },
+    errors: {
+    }
+  }
+
+  handleInputChange = evt => {
+    const date = this.state.date
+    date[evt.target.id] = parseInt(evt.target.value, 10)
+    const errors = validate(date)
+    this.setState({date, errors})
+  }
+
+// If we get a date as a prop we assign it to the state
+  componentWillMount () {
+    this.props.date && this.setState({
```

Comment:
I think if we are going to use internal state, we should initialize it in the constructor. I think we can call setState from this lifecycle method since this is called before render, but it might have some unexpected behavior